test/unit/FutarchyChallenge.js

```diff
@@ -0,0 +1,259 @@
+  const STAKE_AMOUNT          = 10 * 10 ** 18
+  const TRADING_PERIOD        = 60 * 60 * 24
+  const LOWER_BOUND           = 0
+  const UPPER_BOUND           = 100
+  const LMSR_ADDR             = mockAddresses[1]
+  const SCALAR_P_ORACLE_ADDR  = mockAddresses [2]
+
+  let token, challenge, futarchyOracleMock, futarchyOracleFactoryMock
+
+  describe('when deployed correctly, ', () => {
+    before(async () => {
+      token = await Token.new(1000 * 10 ** 18, "FCR Token", 18, 'FCR')
+      futarchyOracleFactoryMock = await newFutarchyOracleFactoryMock()
+      challenge = await deployChallenge({token, futarchyOracleFactory: futarchyOracleFactoryMock})
+    })
+
+    it('sets the correct token address', async () => {
+      expect(await challenge.token()).to.equal(token.address)
+    })
+
+    it('sets the correct listing owner', async () => {
+      expect(await challenge.listingOwner()).to.equal(LISTING_OWNER)
+    })
+
+    it('sets the correct registry address', async () => {
+      expect(await challenge.registry()).to.equal(REGISTRY_ADDR)
+    })
+
+    it('sets the correct challenger', async () => {
+      expect(await challenge.challenger()).to.equal(CHALLENGER)
+    })
+
+    it('sets the correct stake amount', async () => {
+      expect((await challenge.stakeAmount()).toNumber()).to.equal(STAKE_AMOUNT)
+    })
+
+    it('sets the correct trading period', async () => {
+      expect((await challenge.tradingPeriod()).toNumber()).to.equal(TRADING_PERIOD)
+    })
+
+    it('sets the correct upperBound', async () => {
+      expect((await challenge.upperBound()).toNumber()).to.equal(UPPER_BOUND)
+    })
+
+    it('sets the correct lowerBound', async () => {
+      expect((await challenge.lowerBound()).toNumber()).to.equal(LOWER_BOUND)
+    })
+
+    it('sets the correct futarchyOracleFactory', async () => {
+      expect(await challenge.futarchyOracleFactory()).to.equal(futarchyOracleFactoryMock.address)
+    })
+
+    it('sets the correct scalarPriceOracle', async () => {
+      expect(await challenge.scalarPriceOracle()).to.equal(SCALAR_P_ORACLE_ADDR)
+    })
+
+    it('sets the correct lsmrtMarketMaker', async () => {
+      expect(await challenge.lmsrMarketMaker()).to.equal(LMSR_ADDR)
+    })
+
+    it('sets isStarted to false', async () => {
+      expect(await challenge.isStarted()).to.equal(false)
+    })
+
+    it('sets isFunded to false', async () => {
+      expect(await challenge.isFunded()).to.equal(false)
+    })
+  })
+
+  describe('when deployed with incorrect parameters, ', () => {
+    it('throws if tradingPeriod < 1', async () => {
+      let tradingPeriod = 0
+      await expectRevert(deployChallenge({tradingPeriod}))
+    })
+
+    it('throws if upperBound < lowerBound', async () => {
+      let upperBound = 1
+      let lowerBound = 2
+      await expectRevert(deployChallenge({upperBound, lowerBound}))
+    })
+  })
+
+  describe('ended()', async () => {
+    describe('contingent on futarchOracle outcome', async () => {
```

Comment:
spelling: futarchyOracle

Reply:
@emaG3m I might reorganize these a bit .. might want to use the format `describe('when FutarchyOracle outcome is set'` ... `it('returns true'`

Reply:
I think in that case I don't even need a second describe --^  bc there would only be one test per describe. I can do that